### ISSUE TITLE
add timestamp to context tracking amqp payload

### ISSUE THREAD

**Issue by nerdvegas** [nerdvegas/rez#535]
Currently, if one were to write a daemon to consume rez context tracking AMQP messages (to dump into ELK for eg), there is no way to know the original timestamp indicating the actual resolve time. I believe RabbitMQ does have a plugin available to add this info into the message; however, rez does also publish these payloads in a separate thread, so even the message publish time does not necessarily reflect the actual resolve time.

**Comment by JeanChristopheMorinRodeoFX** [comment]
Isn't the timestamp generated from the ResolvedContext objkect itself? And then dumped into the context key of the payload? At least that's what I see on my side. Or I'm missing something...

**Comment by nerdvegas** [comment]
There are two different cases - a context getting created (ie a resolve
occurs), or a context being sourced (ie an existing rxt was used).

In the created case, 'timestamp' does not necessarily match the time of the
resolve. The resolve itself may be timestamped to an earlier date (ie
`rez-env -t`). In this sense 'timestamp' refers to the time as far as the
solver is concerned, which may or may not match current time.

In the sourced case, the timestamp means the same as above... so even if it
_is_ the time of the resolve, that's now sometime in the past, because the
context is just getting reused.

Hth
A

**Comment by JeanChristopheMorinRodeoFX** [comment]
Makes sense. BTW I just saw your edit in the issue description and having the command would indeed make perfect sense. I have setup a queue to dump the events in elasticsearch and I too had the feeling having the command where the resolves come from would be quite useful.
